extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/GeneratedStaticResourceBuildItem.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+
+/**
+ * GeneratedStaticResourceBuildItem aims to be used by extensions to generate static resources. Those resources will be served
+ * on the given publicPath. It is NOT necessary to create the file on disk. Behind the scenes Quarkus will take care of add
+ * those resources to the final build, through {@link AdditionalStaticResourceBuildItem}, {@link NativeImageResourceBuildItem}
+ * and
+ * {@link io.quarkus.deployment.builditem.GeneratedResourceBuildItem} build items.
+ */
+public final class GeneratedStaticResourceBuildItem extends MultiBuildItem {
+
+    private final String publicPath;
```

Comment:
I think we should probably stick to `path`. I don't think we have this notion of `public` elsewhere (but please correct me if I'm wrong).
I see it has been part of the initial proposal but I think it's a bit incorrect as I suppose it might not be entirely public if security is added on top of it.

I would use `path` and copy the paragraph we have above:
```
 * The value of {@code path} should be prefixed with {@code '/'} and is assumed to be a path
 * under {@code 'META-INF/resources'}.
```